File: Resueltos/Luciano_Chavarria/Java/Train Swapping.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        for (int i = 0; i < a; i++) {
            int b = sc.nextInt();
            int m = 0;
            int arr[] = new int[b];
            for (int j = 0; j < b; j++) {
                arr[j] = sc.nextInt();
            }
            for (int j = 0; j < arr.length - 1; j++) {
                for (int k = 0; k < arr.length - 1; k++) {
                    if (arr[k] > arr[k + 1]) {
                        int temp = arr[k];
                        arr[k] = arr[k + 1];
                        arr[k + 1] = temp;
                        m++;
                    }
                }
            }
            System.out.println("Optimal train swapping takes " + m + " swaps.");
        }
    }

}

File: Resueltos/Cristian Maldonado/Train_Swapping.cpp
#include<iostream>
using namespace std;

int cambios (int arr[], int size){
	int contador = 0;
	if(size == 2){
		return 1;
	}
	for(int i = 0; i < size; i++){
		for(int j = 0; j < size-1; j++){
			if(arr[j] > arr[j+1]){
				contador ++;
				int temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
	}
	return contador;
}

int main (int argc, char *argv[]) {
	int casos = 0;
	scanf("%i", &casos);
	for(int i = 0; i < casos; i++){
		int tam = 0;
		scanf("%i", &tam);
		int vector[tam];
		for(int j = 0; j < tam ; j++){
			scanf("%i", &vector[j]);
		}
		int resultado = cambios(vector, sizeof(vector)/sizeof(0));
		cout << "Optimal train swapping takes "+std::to_string(resultado)+" swaps."<<endl;
	}
	return 0;
}
